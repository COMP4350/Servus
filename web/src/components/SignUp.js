import React from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import { Button, TextField } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    margin: '10%'
  },
  textField : {
    width: '50%',
    marginBottom: '3%'
  },
  button : {
    marginBottom: '3%'
  }
}));
const SignUp = () => {
  const classes = useStyles();
  const history = useHistory();
  const validate = () => {
    history.push('/home');
    // validate login
  };
  const createAccount = () => {
    // call the api to create account
  };

  return ( <div className={classes.root}>
<h2>Create Account</h2>
<TextField className={classes.textField}
        id="outlined-basic"
        label="First Name"
      />
      <TextField className={classes.textField}
        id="outlined-basic"
        label="Last Name"
      />
        <TextField className={classes.textField}
        id="outlined-basic"
        label="Username"
      />
      <TextField className={classes.textField}
        id="outlined-password-input"
        label="password"
        type="Password"
        autoComplete="current-password"
      />
      <TextField className={classes.textField}
        id="outlined-password-input"
        label="Confirm Password"
        type="password"
        autoComplete="current-password"
      />
      <Button className={classes.button}
        variant="contained"
        onClick={validate}>
        CREATE ACCOUNT
      </Button>
    </div>)};

    export default SignUp;