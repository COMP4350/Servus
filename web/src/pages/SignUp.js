import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { Button, TextField } from '@material-ui/core';
import useForm from '../hooks/useForm';
import axios from 'axios';

const useStyles = makeStyles(() => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        margin: '10%',
    },
    textField: {
        width: '50%',
        marginBottom: '3%',
    },
    button: {
        marginBottom: '3%',
    },
}));

const SignUp = () => {
    const classes = useStyles();
    const history = useHistory();
    const [errors, setErrors] = useState({});
    const [formValid, setFormValid] = useState();
    const [signUpForm, onFormChange] = useForm({
        username: '',
        password: '',
        confirmPassword: '',
        firstName: '',
        lastName: '',
    });
    const validate = () => {
        let errors = {};
        if (!signUpForm.firstName) {
            errors.username = 'first name is required';
        }
        if (!signUpForm.lastName) {
            errors.username = 'last name is required';
        }
        if (!signUpForm.username) {
            errors.username = 'username is required';
        }
        if (!signUpForm.password) {
            errors.password = 'password is required';
        }
        if (!signUpForm.confirmPassword) {
            errors.confirmPassword = 'confirm password is required';
        } else if (signUpForm.confirmPassword !== signUpForm.password) {
            errors.confirmPassword = 'passwords must match';
        }
        setErrors(errors);
        setFormValid(Object.getOwnPropertyNames(errors).length == 0);
    };
    const createAccount = async () => {
        if (formValid) {
            axios
                .post(
                    `/user/${signUpForm.username}`,
                    {
                        password: signUpForm.password,
                        firstName: signUpForm.firstName,
                        lastName: signUpForm.lastName,
                    },
                    {
                        withCredentials: true,
                    }
                )
                .then(() => {
                    axios.post(
                        `/user/${signUpForm.username}/login`,
                        {
                            password: signUpForm.password,
                        },
                        {
                            withCredentials: true,
                        }
                    );
                    history.push('/');
                })
                .catch(err => {
                    if (err.response.status == 422)
                        alert('username already exists');
                    else alert(err);
                    setFormValid(false);
                });
            return;
        }
    };
    useEffect(() => {
        createAccount();
    }, [formValid]);

    return (
        <div className={classes.root}>
            <h2>Create Account</h2>
            <TextField
                className={classes.textField}
                id="outlined-basic"
                label="First Name"
                name="firstName"
                value={signUpForm.firstName}
                onChange={onFormChange}
                error={errors.firstName}
                helperText={errors.firstName}
            />
            <TextField
                className={classes.textField}
                id="outlined-basic"
                label="Last Name"
                name="lastName"
                value={signUpForm.lastName}
                onChange={onFormChange}
                error={errors.lastName}
                helperText={errors.lastName}
            />
            <TextField
                className={classes.textField}
                id="outlined-basic"
                label="Username"
                name="username"
                value={signUpForm.username}
                onChange={onFormChange}
                error={errors.username}
                helperText={errors.username}
            />
            <TextField
                className={classes.textField}
                id="outlined-password-input"
                label="password"
                type="Password"
                autoComplete="current-password"
                name="password"
                value={signUpForm.password}
                onChange={onFormChange}
                error={errors.password}
                helperText={errors.password}
            />
            <TextField
                className={classes.textField}
                id="outlined-password-input"
                label="Confirm Password"
                type="password"
                name="confirmPassword"
                value={signUpForm.confirmPassword}
                onChange={onFormChange}
                error={errors.confirmPassword}
                helperText={errors.confirmPassword}
            />
            <Button
                className={classes.button}
                variant="contained"
                onClick={validate}>
                CREATE ACCOUNT
            </Button>
        </div>
    );
};

export default SignUp;
