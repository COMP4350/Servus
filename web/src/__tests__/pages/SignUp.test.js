import React from 'react';
import { act, render } from '@testing-library/react';
import SignUp from '../../pages/SignUp';

describe('SignUp', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();

        const { asFragment } = render(<SignUp />);

        expect(asFragment()).toMatchSnapshot();

        await act(() => promise);
    });
});
