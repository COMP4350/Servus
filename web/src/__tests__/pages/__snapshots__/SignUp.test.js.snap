// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <h2>
      Create Account
    </h2>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        First Name
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-basic"
          name="firstName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Last Name
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-basic"
          name="lastName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Username
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-basic"
          name="username"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-password-input"
        id="outlined-password-input-label"
      >
        password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          autocomplete="current-password"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-password-input"
          name="password"
          type="Password"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="outlined-password-input"
        id="outlined-password-input-label"
      >
        Confirm Password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="outlined-password-input"
          name="confirmPassword"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        CREATE ACCOUNT
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
